department-of-veterans-affairs/va.gov-cms
# VACMS-11495: added content search for content audit

## Description

Relates to #11495.
Relates to #11208.
Relates to #11647.

## Testing done


## Screenshots


## QA steps
~Go to Content, and in the content navigation go to Content Audit Tools. Under that you should find a Content Search tab. You can search by keywords like VRRAP, ID.me and Air Force. You can also filter by content type depending on the nature of the audit. ~

- Go to https://pr11706-vs5cfdcsh2vti4crwdkxp7t14onom0hv.ci.cms.va.gov/admin/content/audit/search
- [x] validate that the tab is in the right order (alph order) under Content Audit Tools
- go to [this link](https://pr11706-vs5cfdcsh2vti4crwdkxp7t14onom0hv.ci.cms.va.gov/admin/content/audit/search?keywords=VRRAP&label=&type%5B%5D=banner&type%5B%5D=basic_landing_page&type%5B%5D=campaign_landing_page&type%5B%5D=centralized_content&type%5B%5D=checklist&type%5B%5D=documentation_page&type%5B%5D=event&type%5B%5D=event_listing&type%5B%5D=faq_multiple_q_a&type%5B%5D=full_width_banner_alert&type%5B%5D=health_care_local_facility&type%5B%5D=health_care_local_health_service&type%5B%5D=health_care_region_detail_page&type%5B%5D=health_care_region_page&type%5B%5D=health_services_listing&type%5B%5D=landing_page&type%5B%5D=leadership_listing&type%5B%5D=locations_listing&type%5B%5D=media_list_images&type%5B%5D=media_list_videos&type%5B%5D=nca_facility&type%5B%5D=news_story&type%5B%5D=office&type%5B%5D=outreach_asset&type%5B%5D=page&type%5B%5D=person_profile&type%5B%5D=press_release&type%5B%5D=press_releases_listing&type%5B%5D=promo_banner&type%5B%5D=publication_listing&type%5B%5D=q_a&type%5B%5D=regional_health_care_service_des&type%5B%5D=step_by_step&type%5B%5D=story_listing&type%5B%5D=support_resources_detail_page&type%5B%5D=support_service&type%5B%5D=vamc_operating_status_and_alerts&type%5B%5D=vamc_system_billing_insurance&type%5B%5D=vamc_system_medical_records_offi&type%5B%5D=vamc_system_policies_page&type%5B%5D=vamc_system_register_for_care&type%5B%5D=va_form&type%5B%5D=vba_facility&type%5B%5D=vba_facility_service&type%5B%5D=vet_center&type%5B%5D=vet_center_cap&type%5B%5D=vet_center_facility_health_servi&type%5B%5D=vet_center_locations_list&type%5B%5D=vet_center_mobile_vet_center&type%5B%5D=vet_center_outstation&type%5B%5D=vha_facility_nonclinical_service)
- [x] validate that there are  nodes that show up in the report.
- validate that the View meets the apprproprate ACs from our [Views Style Guide](https://va-gov.atlassian.net/wiki/spaces/VAGOV/pages/1740472321/CMS+Views+Style+Guide).
- [x] Display machinename, not `default` or 'page'
- [x] URL [`/admin/content/{{parent-view-name}}/name-of-view`] /admin/content/audit/content-search
- [x] Menu item setting: Is it where it belongs?
- [x] Limit to 100 per page, paginated
- [x] Show result count in header. (`<p>Displaying @start - @end of @total</p>`)
- [x] Has table caption describing the table
- [ ] Columns
  - [x] Title (sortable) 
  - [x] Edit button (not operations, no heading) 
  - [x] Content type ~(sortable) (if applicable)~
  - [x] Moderation State
  - [x] Section (field_administration) ~(sortable)~
  - [ ] ~Is Locked (optional)~
  - [ ] Last updated (date time and by user) (sortable, default sort, DESC)
  - [ ] [Other content fields as needed] (sortable)
  - [ ] Exposed
    - [x] "Title contains" (text search of title field)
    - [x] "Content type" (single value drop down or multivalue? should options be limited) 
    - [x] "Moderation state"  (single value drop down or multivalue?)
    - [ x] "Section" ~(workbench access section; single select, show children)~ Added:  due to the nature of search api, this is not attainable as a select list./
    - [x] Order of filters matches order of columns
    - [x] Reset button exists
    - [x] Apply button says 'Filter' 
- [x] Tab navigation is alpha sorted or intentionally grouped
- [ ] Not exposed
    - [x] only by relevance

**Added 12/12**
- [ ] vist https://pr11706-vs5cfdcsh2vti4crwdkxp7t14onom0hv.ci.cms.va.gov/admin/content/audit/rich-text-field-audit  validated you are directed from the old view to the new
- [x] validate old "rich text" tab no longer appears under audit.
- [x] visit search for [diabetic care in terms](https://pr11706-vs5cfdcsh2vti4crwdkxp7t14onom0hv.ci.cms.va.gov/admin/content/audit/search?search=diabetic+care&title=&etype_op=or&etype=entity%3Ataxonomy_term&ms=All&published=All&section=)  validate  several terms show up.
   - [x] Click the download csv button validate that it saves a file to your machine.
   - [x] Open it in excell or libreoffice or other spreadsheet.  validate there are the same number of result as the visible view.
   - [x] validate that the edit column contains full urls to prod.
   - [x] validate that the url collumn contains full urls to prod
   - [x] validate that the rows all line up like they are supposed to (no data escaped its colunm)
- [x] Do a [media search for "verterans day"](https://pr11706-vs5cfdcsh2vti4crwdkxp7t14onom0hv.ci.cms.va.gov/admin/content/audit/search?search=veterans+day&title=&etype_op=or&etype=entity%3Amedia&ms=All&published=All&section=)
   - [x] validate that media shows up whose contents contain the search term.
   - [x] validate the edit button takes you to the media edit
   - [x] validate the search excerpt shows up
   - [x] validate the content type column identifies it as media.


added 12/13
- Login as kitty.walsh  
   - [x] Validate that you can see the  https://pr11706-vs5cfdcsh2vti4crwdkxp7t14onom0hv.ci.cms.va.gov/admin/content/audit/search

### Definition of Done


### Select Team for PR review



### Is this PR blocked by another PR?


### Does this PR need review from a Product Owner


### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
